apps/studio/src/lib/editor/extensions/partialReadOnly.ts

```diff
@@ -0,0 +1,424 @@
+import {
```

Comment:
A lot of codes because many of these were handled in cm5, but now we have to track almost everything (checking marker positions, states, etc.) by ourselves lol.

Reply:
Damn this is a lot of code to handle this! Is this general enough to make it a full CM6 plugin that we can commit back to them?

Reply:
Need a little tweak before we can do that though.